crates/wasmparser/src/readers/core/types/matches.rs

```diff
@@ -252,6 +259,40 @@ impl Matches for WithRecGroup<StorageType> {
     }
 }
 
+impl<'a> Matches for WithRecGroup<&'a ContType> {
+    fn matches(types: &TypeList, a: Self, b: Self) -> bool {
+        match (*a, *b) {
+            (ContType(ca), ContType(cb)) => {
+                ca == cb || {
+                    let core_id_a = if let Some(ida) = ca.as_core_type_id() {
+                        ida
+                    } else {
+                        types
+                            .at_canonicalized_unpacked_index(
+                                WithRecGroup::rec_group(a),
+                                ca.unpack(),
+                                usize::MAX,
+                            )
+                            .expect("type references are checked during canonicalization")
+                    };
```

Comment:
This looks like it may be a duplicate of `let core_type_id = |...| ...;` in validator/types.rs, would you be up for extracting that to a helper function on `TypeList` to help share the implementation between these locations?

Reply:
Alternatively what you could also perhaps do here is to manufacture a `RefType`  and then call `reftype_is_subtype_impl` perhaps?

Reply:
Yes! I opted for the second approach.